### DISCUSSION TITLE
Unusual Callback URL behaviour when login dashboard (Netlify)

### DISCUSSION THREAD

**Discussion by pytatbro** [avitorio/outstatic#270] [Q&A]
Hi!

I set up Outstatic for local development, and it is really great. However, when I deploy my site to Netlify, there is an issue in the login process.

So I set up my Github OAuth and pass in these values:

- Homepage URL: https://mydomain.com
- Authorization callback URL: https://mydomain.com/api/outstatic/callback

When I try logging in, sometimes it is successful. However, sometimes it redirects me to https://67535705052609009e19308a--mydomain.netlify.app/outstatic?code=something (should it be https://mydomain.com/outstatic?code=something ?). Because of this, it shows me the login screen again instead of the dashboard. If I manually put https://mydomain.com/outstatic into the browser, then it serves me the dashboard as normal.

I found that https://67535705052609009e19308a--mydomain.netlify.app/ is the Permalink of the lastest published deployment on Netlify.
<img width="588" alt="deploy" src="https://github.com/user-attachments/assets/c95ab602-a6a6-4988-9846-f9d5e9b6d041">

So, I think the issue could be within the Next.js Server Handler function that Netlify uses on all deployed Next.js sites.
I am pretty new to Next.js and honestly haven't tried Vercel before. Is this behavior only applied to Netlify?

That aside, Outstatic is really great so far. I really appreciate the effort you guys put in 👏
Just for reference, I am using Next.js 15.0.3 and Outstatic ^1.4.13

**Comment by avitorio** [comment]
Hey @pytatbro ,

This is probably a Netlify or GitHub specific issue. If your OAuth settings never mention your preview deployment url, how come it redirects you to it? It shouldn't know about that url.

I haven't deployed Outstatic to Netlify before.

Question, are you starting your login from yourdomain.com/outstatic or from (deploy preview url)/outstatic?
What happens when you try to login again on the preview url?

**Reply by pytatbro** [reply]
At first, I started my login from mydomain.com/outstatic. It just automatically redirects to the deploy URL. I also test to use the deploy URL (i.e. start login from deploy URL and set OAuth to deploy URL) then it became even weirder. Sometimes it redirects to someid--mydomain.netlify.app/outstatic, and sometimes it redirects to master--mydomain.netlify.app/outstatic. It happens very randomly.

I tried rebuilding and redeploying, and it even redirected to the new deploy URL (with a different ID). I guess this is specifically related to Netlify. Maybe I will try to contact them.

**Reply by avitorio** [reply]
Hi @pytatbro, did you find a solution?
